purchases/src/main/kotlin/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -502,51 +497,45 @@ internal class BillingWrapper(
-            if (querySubsPurchasesParams == null) {
-                errorLog(PurchaseStrings.INVALID_PRODUCT_TYPE.format("getPurchaseType"))
-                listener(ProductType.UNKNOWN)
-                return
-            }
-
-            client.queryPurchasesAsyncWithTrackingEnsuringOneResponse(
-                BillingClient.ProductType.SUBS,
-                querySubsPurchasesParams,
-            ) querySubPurchasesAsync@{ querySubsResult, subsPurchasesList ->
-
-                val subsResponseOK = querySubsResult.responseCode == BillingClient.BillingResponseCode.OK
-                val subFound = subsPurchasesList.any { it.purchaseToken == purchaseToken }
-                if (subsResponseOK && subFound) {
-                    listener(ProductType.SUBS)
-                    return@querySubPurchasesAsync
-                }
-
-                val queryInAppsPurchasesParams = BillingClient.ProductType.INAPP.buildQueryPurchasesParams()
-                if (queryInAppsPurchasesParams == null) {
-                    errorLog(PurchaseStrings.INVALID_PRODUCT_TYPE.format("getPurchaseType"))
-                    listener(ProductType.UNKNOWN)
-                    return@querySubPurchasesAsync
-                }
-                client.queryPurchasesAsyncWithTrackingEnsuringOneResponse(
-                    BillingClient.ProductType.INAPP,
-                    queryInAppsPurchasesParams,
-                ) queryInAppPurchasesAsync@{ queryInAppsResult, inAppPurchasesList ->
-
-                    val inAppsResponseIsOK = queryInAppsResult.responseCode == BillingClient.BillingResponseCode.OK
-                    val inAppFound = inAppPurchasesList.any { it.purchaseToken == purchaseToken }
-                    if (inAppsResponseIsOK && inAppFound) {
+        queryPurchaseType(BillingClient.ProductType.SUBS, purchaseToken, listener) { subFound ->
+            if (subFound) {
+                listener(ProductType.SUBS)
+            } else {
+                queryPurchaseType(BillingClient.ProductType.INAPP, purchaseToken, listener) { inAppFound ->
+                    if (inAppFound) {
                         listener(ProductType.INAPP)
-                        return@queryInAppPurchasesAsync
+                    } else {
+                        listener(ProductType.UNKNOWN)
                     }
-                    listener(ProductType.UNKNOWN)
-                    return@queryInAppPurchasesAsync
                 }
             }
-        } ?: listener(ProductType.UNKNOWN)
+        }
+    }
+
+    private fun queryPurchaseType(
+        productType: String,
+        purchaseToken: String,
+        listener: (ProductType) -> Unit,
+        resultHandler: (Boolean) -> Unit,
+    ) {
+        QueryPurchasesByTypeUseCase(
+            QueryPurchasesByTypeUseCaseParams(
+                dateProvider,
+                diagnosticsTrackerIfEnabled,
+                appInBackground = false,
```

Comment:
We're hardcoding this to false right now. Are you planning to pass it in on another PR?

Reply:
Oh this is not correct yeah. I will fix it.

Reply:
ok, there's a bit of an issue here...

`getStoreTransaction` gets called from `onPurchasesUpdated`, which is a billing client interface, so we don't have access to `appInBackground`....

I think it's time to inject that `StateProvider`